pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,350 @@
+	response, err := resource.Send()
+	if err != nil {
+		return err
+	}
+	clusters := response.Items()
+	opts.clusterList = *clusters
+	return nil
+}
+
+func runClusterSelectionInteractivePrompt(opts *options) error {
+	clusterListString := make([]string, 0)
+	for _, cluster := range opts.clusterList.Slice() {
+		clusterListString = append(clusterListString, cluster.Name())
+	}
+
+	// TODO add page size and Localizer
+	prompt := &survey.Select{
+		Message: "Select the cluster to register",
+		Options: clusterListString,
+	}
+
+	var selectedClusterName string
+	err := survey.AskOne(prompt, &selectedClusterName)
+	if err != nil {
+		return err
+	}
+
+	// get the desired cluster
+	for _, cluster := range opts.clusterList.Slice() {
+		if cluster.Name() == selectedClusterName {
+			opts.selectedCluster = *cluster
+		}
+	}
+	return nil
+}
+
+func parseDNSURL(opts *options) (string, error) {
+	clusterIngressDNSName := opts.selectedCluster.Console().URL()
+	if len(clusterIngressDNSName) == 0 {
+		return "", fmt.Errorf("url is empty")
+	}
+	return strings.SplitAfter(clusterIngressDNSName, ".apps.")[1], nil
+}
+
+// TODO this function should be split the ocm call and the response flow logic
+func getMachinePoolList(opts *options) error {
+	// ocm client connection
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, err := conn.API().OCMClustermgmt()
+	if err != nil {
+		return err
+	}
+
+	resource := client.Clusters().Cluster(opts.selectedCluster.ID()).MachinePools().List()
+	response, err := resource.Send()
+	if err != nil {
+		return err
+	}
+	if response.Size() == 0 {
+		createMachinePoolInteractivePrompt(opts)
+	} else {
+		for _, machinePool := range response.Items().Slice() {
+			opts.existingMachinePoolList = append(opts.existingMachinePoolList, *machinePool)
+		}
+		err = validateMachinePoolNodes(opts)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func checkForValidMachinePoolLabels(machinePool v1.MachinePool) bool {
+	labels := machinePool.Labels()
+	for key, value := range labels {
+		if key == machinePoolLabelKey && value == "standard" {
+			return true
+		}
+	}
+	return false
+}
+
+func validateMachinePoolCount(count int) bool {
+	if count <= 2 || count%3 != 0 {
+		return false
+	}
+	return true
+}
+
+func checkForValidMachinePoolTaints(machinePool v1.MachinePool) bool {
+	taints := machinePool.Taints()
+	for _, taint := range taints {
+		if taint.Effect() == machinePoolTaintEffect && taint.Key() == machinePoolTaintKey {
+			return true
+		}
+	}
+	return false
+}
+
+func createNewMachinePoolTaintsDedicated() *v1.TaintBuilder {
+	return v1.NewTaint().
+		Key(machinePoolTaintKey).
+		Effect(machinePoolTaintEffect).
+		Value(machinePoolTaintValue)
+}
+
+func createNewMachinePoolLabelsDedicated() (*v1.Label, error) {
+	label := v1.NewLabel().
+		Key(machinePoolLabelKey).
+		Value(machinePoolLabelValue)
+	l, err := label.Build()
+	if err != nil {
+		return nil, err
+	}
+	return l, nil
+}
```

Comment:
This could just return a `map[string]string` to avoid the need for `createMachinePoolLabelMap`